core/src/main/scala/org/apache/spark/BarrierTaskContext.scala

```diff
@@ -80,7 +101,45 @@ class BarrierTaskContext(
   @Experimental
   @Since("2.4.0")
   def barrier(): Unit = {
-    // TODO SPARK-24817 implement global barrier.
+    val callSite = Utils.getCallSite()
+    logInfo(s"Task $taskAttemptId from Stage $stageId(Attempt $stageAttemptNumber) has entered " +
+      s"the global sync, current barrier epoch is $barrierEpoch.")
+    logTrace(s"Current callSite: $callSite")
```

Comment:
How about the following since `Utils.getCallSite()` is not lightweight?
```
if (isTraceEnabled) {
  val callSite = Utils.getCallSite()
  logTrace(s"Current callSite: $callSite")
}
```

Reply:
+1

Reply:
or simpler: `logTrace("Current callSite: " + Utils.getCallSite())`